core/src/main/java/org/apache/iceberg/BaseMetastoreTableOperations.java

```diff
@@ -200,9 +200,18 @@ protected void refreshFromMetadataLocation(String newLocation, Predicate<Excepti
   private String metadataFileLocation(TableMetadata metadata, String filename) {
     String metadataLocation = metadata.properties()
         .get(TableProperties.WRITE_METADATA_LOCATION);
+    String pinterestMetadataPrefix = metadata.properties()
```

Comment:
Why don't we just set `TableProperties.WRITE_METADATA_LOCATION` at catalog level and add additional property to selective suffix this location with db/tbl info in path.